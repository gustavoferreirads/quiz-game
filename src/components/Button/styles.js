import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';

const styles = StyleSheet.create({
  container: {
    height: 40,
    borderRadius: 30,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 20,
    fontWeight: 'bold',
    width: 150,
  },

  'button-inline': {
    backgroundColor: colors.primary,
  },

  'button-outline': {
    backgroundColor: '#FFF',
    borderWidth: 1,
    borderColor: colors.primary,
  },

  text: {
    fontSize: fonts.big,
  },
  'text-inline': {
    color: '#FFF',
  },
  'text-outline': {
    color: colors.primary,
  },
});

export default styles;
