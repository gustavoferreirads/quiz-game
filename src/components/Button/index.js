import React, { Fragment } from 'react';
import * as PropTypes from 'prop-types';
import { Text, TouchableOpacity } from 'react-native';
import styles from './styles';

const Button = ({ style, children, type, onPress }) => (
  <TouchableOpacity style={[styles.container, style, styles[`button-${type}`]]} onPress={onPress}>
    <Text style={[styles.text, styles[`text-${type}`]]}>{children}</Text>
  </TouchableOpacity>
);

Button.defaultProps = {
  type: 'inline',
  onPress: () => {},
  style: {},
  children: <Fragment />,
};

Button.propTypes = {
  children: PropTypes.element,
  style: PropTypes.object,
  type: PropTypes.string,
  onPress: PropTypes.func,
};

export default Button;
