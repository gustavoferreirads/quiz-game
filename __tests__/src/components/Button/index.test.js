import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import Button from '../../../../src/components/Button';

it('renders a Button ', () => {
  expect(
    shallow(
      <Button onPress={jest.fn} style={{ backgroundColor: 'red' }}>
        <Text>Label</Text>
      </Button>,
    ),
  ).toMatchSnapshot();
});

it('renders a Button with style "inline"', () => {
  expect(shallow(<Button type="inline" />)).toMatchSnapshot();
});

it('renders a Button with style "outline"', () => {
  expect(shallow(<Button type="inline" />)).toMatchSnapshot();
});

it('renders a Button with style nonexistent ', () => {
  expect(shallow(<Button type="quizStyle" />)).toMatchSnapshot();
});
