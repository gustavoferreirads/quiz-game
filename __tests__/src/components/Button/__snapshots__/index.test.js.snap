// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Button  1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "fontWeight": "bold",
        "height": 40,
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": 150,
      },
      Object {
        "backgroundColor": "red",
      },
      Object {
        "backgroundColor": "#FF9982",
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    <Component>
      Label
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`renders a Button with style "inline" 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "fontWeight": "bold",
        "height": 40,
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": 150,
      },
      Object {},
      Object {
        "backgroundColor": "#FF9982",
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  />
</TouchableOpacity>
`;

exports[`renders a Button with style "outline" 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "fontWeight": "bold",
        "height": 40,
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": 150,
      },
      Object {},
      Object {
        "backgroundColor": "#FF9982",
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  />
</TouchableOpacity>
`;

exports[`renders a Button with style nonexistent  1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "fontWeight": "bold",
        "height": 40,
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": 150,
      },
      Object {},
      undefined,
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        undefined,
      ]
    }
  />
</TouchableOpacity>
`;
